src/kemal/param_parser.cr

```diff
@@ -57,6 +62,23 @@ module Kemal
       end
     end
 
+    def parse_file_upload
+      HTTP::FormData.parse(@request) do |field, data, meta, headers|
+        if !meta.nil?
+          filename = meta.filename
+          if !filename.nil?
+            tempfile = Tempfile.new(filename)
+            ::File.open(tempfile.path, "w") do |file|
+              IO.copy(data, file)
+            end
+            @files[field] = FileUpload.new(tmpfile: tempfile, tmpfile_path: tempfile.path, meta: meta, headers: headers)
+          else
+            @body[field] = data.gets_to_end
```

Comment:
This looks like an easy DOS, allowing an attacker to read an unlimited size into memory. I would suggest providing a conservative length limit to every memory read op.